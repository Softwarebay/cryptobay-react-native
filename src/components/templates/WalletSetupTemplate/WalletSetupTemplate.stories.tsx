import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';

import { WalletSetupTemplate } from './WalletSetupTemplate';

storiesOf('Templates/WalletSetupTemplate', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Wallet Setup Template', () => (
    <WalletSetupTemplate
      onImportUsingSeedPhrase={action('onImportUsingSeedPhrase')}
      onCreateANewWallet={action('onCreateANewWallet')}
    />
  ));
