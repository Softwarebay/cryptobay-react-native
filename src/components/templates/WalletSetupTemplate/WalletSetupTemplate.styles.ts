import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  walletSetupTempate: {
    flex: 1,
    width: '100%',
    justifyContent: 'center',
    backgroundColor: '#17171A',
  },
  contentWrapper: {
    padding: 30,
    width: '100%',
  },
  picture: {
    alignSelf: 'center',
    marginBottom: 80,
  },
  header: {
    marginBottom: 40,
  },
  importButton: {
    marginBottom: 16,
  },
});
