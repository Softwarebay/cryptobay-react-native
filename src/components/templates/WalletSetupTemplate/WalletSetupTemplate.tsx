import React from 'react';
import { SafeAreaView, View } from 'react-native';

import { SETUP } from '../../../../assets/pictures';
import { Header, Picture } from '../../atoms';
import { Button } from '../../molecules';

import { styles } from './WalletSetupTemplate.styles';

type WalletSetupTemplateProps = {
  onImportUsingSeedPhrase: () => void;
  onCreateANewWallet: () => void;
};

export const WalletSetupTemplate: React.FC<WalletSetupTemplateProps> = ({
  onImportUsingSeedPhrase,
  onCreateANewWallet,
}: WalletSetupTemplateProps) => {
  return (
    <SafeAreaView style={styles.walletSetupTempate}>
      <View style={styles.contentWrapper}>
        <Picture style={styles.picture} pictureFile={SETUP} />
        <Header style={styles.header} value="Wallet setup" />
        <Button
          testID="wallet-setup-import"
          style={styles.importButton}
          label="Import Using Seed Phrase"
          onPress={onImportUsingSeedPhrase}
        />
        <Button
          testID="wallet-setup-create-new"
          label="Create a New Wallet"
          variant="bright"
          onPress={onCreateANewWallet}
        />
      </View>
    </SafeAreaView>
  );
};
