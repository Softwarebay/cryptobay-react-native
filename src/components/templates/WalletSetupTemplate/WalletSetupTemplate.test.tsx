import React from 'react';
import '@testing-library/jest-native';
import 'react-native';
import renderer from 'react-test-renderer';

import { WalletSetupTemplate } from './WalletSetupTemplate';

const mockOnImportUsingSeedPhrase = jest.fn();
const mockOnCreateANewWallet = jest.fn();

describe('WalletSetupTemplate', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <WalletSetupTemplate
          onImportUsingSeedPhrase={mockOnImportUsingSeedPhrase}
          onCreateANewWallet={mockOnCreateANewWallet}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
