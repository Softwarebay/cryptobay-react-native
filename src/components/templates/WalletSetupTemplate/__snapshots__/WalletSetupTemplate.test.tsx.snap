// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletSetupTemplate Component Snapshot 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#17171A",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "padding": 30,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/pictures/setup.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "marginBottom": 80,
        }
      }
    />
    <RNCMaskedView>
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
                "lineHeight": 56,
              },
              Object {
                "fontSize": 40,
                "fontWeight": "400",
              },
              Object {
                "marginBottom": 40,
              },
            ]
          }
        >
          Wallet setup
        </Text>
      </View>
      <BVLinearGradient
        colors={
          Array [
            4289318399,
            4285724369,
            4288736653,
            4294890341,
            4294616222,
          ]
        }
        endPoint={
          Object {
            "x": 1,
            "y": 0,
          }
        }
        locations={null}
        startPoint={
          Object {
            "x": 0,
            "y": 0,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
                "lineHeight": 56,
              },
              Object {
                "fontSize": 40,
                "fontWeight": "400",
              },
              Object {
                "opacity": 0,
              },
              Object {
                "marginBottom": 40,
              },
            ]
          }
        >
          Wallet setup
        </Text>
      </BVLinearGradient>
    </RNCMaskedView>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2A2D3C",
          "borderRadius": 8,
          "color": "#FEBF32",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 16,
          "opacity": 1,
          "padding": 16,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins",
            },
            Object {
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 24,
            },
            Object {
              "color": "#FEBF32",
            },
          ]
        }
      >
        Import Using Seed Phrase
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FEBF32",
          "borderRadius": 8,
          "color": "#000000",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "padding": 16,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins",
            },
            Object {
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 24,
            },
            Object {
              "color": "#000000",
            },
          ]
        }
      >
        Create a New Wallet
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
